iac-extensions/kubernetes/src/main/java/org/sonar/iac/helm/tree/utils/GoTemplateAstUtils.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.helm.tree.utils;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.sonar.iac.helm.ListNode;
+import org.sonar.iac.helm.NumberNode;
+import org.sonar.iac.helm.StringNode;
+import org.sonar.iac.helm.tree.Node;
+
+public class GoTemplateAstUtils {
+  private GoTemplateAstUtils() {
+    // utility class
+  }
+
+  public static Node unpackNode(Any nodePb) throws InvalidProtocolBufferException {
+    var className = nodePb.getTypeUrl().substring(nodePb.getTypeUrl().lastIndexOf('/') + 1);
+    AnyToNodeConverter converter;
+    switch (className) {
```

Comment:
Probably there is no way to avoid this huge `switch`?

Reply:
I implemented something with reflection, should be safe because we know for sure what we put as types into `Any` nodes

Reply:
Changed to use a map for all known types